verifier/verifier_test.go

```diff
@@ -277,10 +306,104 @@ func (vt *verifyTest) checkVerifyResult(res *VerificationResult) error {
 	if err := vt.compareParentSPKISubjectToParents(res); err != nil {
 		return err
 	}
+
+	if vt.InRevocationSet != res.InRevocationSet {
+		return fmt.Errorf("unexpected InRevocationSet: %t", res.InRevocationSet)
+	}
 	return nil
 }
 
 var verifyTests = []verifyTest{
+	{
+		Name:      "PEMLEX3SignedByISRGRootX1-in-CRLSet",
+		Leaf:      data.PEMDAdrianIOSignedByLEX3, // idx=0
+		Presented: nil,
+		Intermediates: []string{
+			data.PEMLEX3SignedByDSTRootCAX3, // idx=1
+			data.PEMLEX3SignedByISRGRootX1,
+		},
+		Roots: []string{
+			data.PEMDSTRootCAX3SignedBySelf, // idx=3
+		},
+		CurrentTime: 1501804800, // 2017-08-04T00:00:00
+		ExpectedChains: [][]int{
+			{0, 1, 3},
```

Comment:
It looks like we still add chains that are rejected by a CRLSet / OneCRL to `ExpectedChains`. Is that intended behavior?

Reply:
@dadrian that is consistent with existing behaviour when a cert is revoked via CRL or OCSP.
I checked `OCSPRevoked` and `CRLRevoked` , and they do not affect currently `ExpectedChains` if a cert is revoked.

I guess, if we decide to change this behaviour, it should be done in a separate PR along with `OCSPRevoked` and `CRLRevoked`.

@zakird FYI
let's discuss

Reply:
In the code, the chains are built before the revocation checks are performed:
```
	allChains = append(allChains, res.CurrentChains...)
	allChains = append(allChains, res.ExpiredChains...)
	allChains = append(allChains, res.NeverValidChains...)

        if opts.OneCRL != nil {}
        if opts.ShouldCheckOCSP {}
        if opts.ShouldCheckCRL {}
```